1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,359 @@
+completions. Predictive IntelliSense includes an extension model to support the registration of
+additional providers.
+
+Predictions may be displayed in either **InlineView** or **ListView** depending on the user’s
+preference. Below, predictions are displayed with InlineView.
+
+![image](./media/pred-wlc.png)
+
+Below, predictions are displayed with a dropdown in ListView.
+
+![image](./media/pred-listview.png)
+
+## Motivation
+
+Tab Completion has accelerated the success of both new and experienced PowerShell users. New users
+get the benefit of discovery; seeing available cmdlets and parameters as options while interactively
+typing. Experienced users receive the benefit of acceleration; typing less while using the **<tab>** key
+to quickly complete a command.
+
+The increasing amount of technology translates to an increase in cmdlets and full command
+complexity. Predictive IntelliSense is an addition to the concept of Tab Completion; helping the
+user discover, build and edit full commands based on the user’s history or additional plugins.
+
+## Release plan
+
+- Proposed 7.1 GA
+  - Currently in PowerShell 7.1 PReview 7
+  - Predictive IntelliSense Whole Line Completion
+  - Historical Predictions
+  - Extension Framework
+- Future PSReadLine previews
+  - Predictive IntelliSense Whole Line Completion + ListView
+  - Support for additional providers using extension framework
+  - Previews begin in October with PSReadLine 2.2.0-beta1
+
+## Goals/Non-Goals
+
+ Goals:
+
+- Provide suggestions from PSReadLine history
+- The user will be able to enable and disable Predictive IntelliSense
+- The user will be able to change between InlineView and ListView
+- The user will have keyboard shortcuts to navigate and edit a prediction
+- The user will register additional providers when desired
+- The user may customize the color of a prediction to support accessibility needs
+- The user will be able to search history for similar predictions
+
+Non-goals:
+
+- Due to limit use, PowerShell Session history is not part of the historical suggestion
+- VSCode implementation. We agree this is important for a consistent experience and are working to
+  bring this feature in a future release
+
+## Specification
+
+The proposal is to add Predictive IntelliSense and Extension model to improve the user interactive experience.
+
+### Key Design Considerations Predictive IntelliSense
+
+Predictive IntelliSense displays a best match command completion from command history. The
+prediction may change if a better match is found as the user types additional information. The
+predicted command may be accepted, edited, or ignored.
+
+![Prediction - Historical](./media/pre-wlc.png)
+
+### Install and Remove Predictive IntelliSense
+
+Predictive IntelliSense is a part of the PSReadLine module. Updating or installing the current beta
+release from PSGallery will provide the Predictive IntelliSense feature.
+
```

Comment:
Can we add info here about our commitment (or non-commitment) to working down to Windows PS 5.1?

Reply:
Yes - I will add information that the historical predictions in PSReadLine, (not the extension framework or plugin support) will be available to be downloaded from PsGallery for Windows Powershell 5.1